assets/js/arrayProcessingTool.js

```diff
@@ -1,94 +1,187 @@
-const inputArrayHTMLElement=document.getElementById('array');
-const resultHTMLElement=document.getElementById('result');
-const searchButtonHTMLElement=document.getElementById('search-button');
-const sequenceButtonHTMLElement=document.getElementById('sequence-button');
-const maximalSumButtonHTMLElement=document.getElementById('maximal-sum-button');
+const inputArrayHTMLElement = document.getElementById('array');
+const resultHTMLElement = document.getElementById('result');
+const searchButtonHTMLElement = document.getElementById('search-button');
+const sequenceButtonHTMLElement = document.getElementById('sequence-button');
+const maximalSumButtonOn2HTMLElement = document.getElementById('maximal-sum-button-0n2');
+const maximalSumButtonOn1HTMLElement = document.getElementById('maximal-sum-button-0n1');
 
-inputArrayHTMLElement.onchange=()=>{
-  object.arr=inputArrayHTMLElement.value.split(' ').map(el=>Number(el));
-  console.log(object.arr);
-  if(object.arr.includes(NaN)){
-      resultHTMLElement.style.backgroundColor='#f85454'
-      resultHTMLElement.textContent='some element(s) in array is not correct. Result could fail';
-  }
+inputArrayHTMLElement.onchange = () => {
+    objectArrProcessTools.arr = inputArrayHTMLElement.value.trim().split(' ').map(el => Number(el));
+    console.log(objectArrProcessTools.arr);
 };
 
-const object={
-    arr:[],
-    subSum: function (){
+const objectArrProcessTools = {
+    arr: [],
+    arrElCheck: function () {
+        if (this.arr.includes(NaN)) {
+            resultHTMLElement.style.backgroundColor = '#f85454'
+            resultHTMLElement.textContent = 'some element(s) in array is not correct. Result could fail';
+            return false;
+        }
+        resultHTMLElement.style.backgroundColor = '#56e360';
+        return true;
+    },
+    negativeNumbersCheck: function (){
+        const arrTmpM = this.arr.filter(el => el < 0);
+        if (arrTmpM.length === this.arr.length) {
+            let result = this.search().split('').filter(el => Number(el));
+            return -result[0];
+        }
+        return null;
+    },
+    subSumOn1: function () {
+        const negativeNumbersAll=this.negativeNumbersCheck();
+        if(negativeNumbersAll){
+            return negativeNumbersAll;
+        }
+        let arrTmp = this.arr;
+        let maxSum = 0;
+        let currSum = 0;
+        for (let i = 0; i < arrTmp.length; i++) {
+            if (currSum + arrTmp[i] >= 0) {
+                currSum += arrTmp[i];
+            }
+            if (currSum > maxSum) {
+                maxSum = currSum;
+            }
+        }
+        return maxSum;
+    },
+    subSumOn2: function () {
```

Comment:
The program shows the invalid result with the next input data:

- -1 -2 -3 (should be 0)

Reply:
fixed